chains/orchestrator-relays/runtime/dancelight/src/tests/common/xcm/bridge_to_eth.rs

```diff
@@ -0,0 +1,151 @@
+                1,
+                msg_size
+            )
+        );
+    });
+
+    // Force process bridge messages
+    mocknets::force_process_bridge::<Dancelight, DancelightPara>();
+
+    // xcm command generated by root_test_send_msg_to_eth
+    let payload = sp_core::blake2_256((nonce, 0).encode().as_ref()).to_vec();
+    let command = Command::Test(payload);
+
+    // msg sent in bridge
+    let msgs = mocknets::eth_bridge_sent_msgs();
+    let sent_message = msgs.first().unwrap();
+
+    assert_eq!(
+        sent_message.channel_id,
+        snowbridge_core::PRIMARY_GOVERNANCE_CHANNEL
+    );
+    assert_eq!(sent_message.command, command.index());
+    assert_eq!(sent_message.params, command.abi_encode());
+}
+
+#[test]
+fn receive_msg_from_eth_validators_are_updated() {
+    Dancelight::execute_with(|| {
+        let origin =
+            <Runtime as frame_system::Config>::RuntimeOrigin::signed(DancelightSender::get());
+
+        // New validators to be added
+        let payload_validators = vec![
+            AccountKeyring::Alice.to_account_id(),
+            AccountKeyring::Charlie.to_account_id(),
+            AccountKeyring::Bob.to_account_id(),
+        ];
+
+        let payload = Payload {
+            magic_bytes: MAGIC_BYTES,
+            message: SymbioticMessage::V1(InboundCommand::<Runtime>::ReceiveValidators {
+                validators: payload_validators.clone(),
+                timestamp: 0u64,
+            }),
+        };
+
+        let event = OutboundMessageAccepted {
+            channel_id: <[u8; 32]>::from(PRIMARY_GOVERNANCE_CHANNEL).into(),
+            nonce: 1,
+            message_id: Default::default(),
+            payload: payload.encode(),
+        };
+
+        let message = Message {
+            event_log: Log {
+                // gateway address
+                address: <Runtime as snowbridge_pallet_inbound_queue::Config>::GatewayAddress::get(
+                ),
+                topics: event
+                    .encode_topics()
+                    .into_iter()
+                    .map(|word| H256::from(word.0 .0))
+                    .collect(),
+                data: event.encode_data(),
+            },
+            proof: mock_snowbridge_message_proof(),
+        };
+
+        // Submit message to the queue
+        assert_ok!(
+            <Dancelight as DancelightRelayPallet>::EthereumInboundQueue::submit(origin, message)
+        );
+
+        let whitelisted =
+            <Dancelight as DancelightRelayPallet>::ExternalValidators::whitelisted_validators();
+
+        // Ignore whitelisted
+        let mut new_validators =
+            <Dancelight as DancelightRelayPallet>::ExternalValidators::validators()
+                .into_iter()
+                .filter(|v| !whitelisted.contains(v))
+                .collect::<Vec<_>>();
+
+        let mut expected_validators = payload_validators.clone();
+
+        new_validators.sort();
+        expected_validators.sort();
```

Comment:
The pallet doesn't change the order, it stores validators in the same order as the received message, so maybe this sort is not needed